main/lsp/LSPPreprocessor.cc

```diff
@@ -119,121 +98,179 @@ bool LSPPreprocessor::ensureInitialized(LSPMethod method, const LSPMessage &msg)
     return false;
 }
 
-void LSPPreprocessor::preprocessAndEnqueue(QueueState &state, unique_ptr<LSPMessage> msg, absl::Mutex &stateMtx) {
-    ENFORCE(owner == this_thread::get_id());
-    if (msg->isResponse()) {
-        return;
-    }
-
-    const LSPMethod method = msg->method();
-    if (!ensureInitialized(method, *msg)) {
-        // msg is invalid. Error sent to client.
-        return;
+unique_ptr<LSPTask> LSPPreprocessor::getTaskForMessage(LSPMessage &msg) {
```

Comment:
Provides the 1:1 mapping between each LSPMessage and an LSPTask. It always returns an LSPTask.